docs/tutorials/intro-to-rust.md

```diff
@@ -0,0 +1,574 @@
+
+<blockquote class="info">
+  "…most functions will end up being inside impl blocks…"
+  <a href="https://doc.rust-lang.org/std/keyword.fn.html" target="_blank">- Rust docs</a>
+</blockquote>
+
+```rust
+let mut quit = false;
+    while !quit {
+        println!("Please type 'increment' (i), 'decrement' (d), or 'get_num' (get) to call those functions.");
+        // clear old value and read the standard input from user
+        input.clear();
+        io::stdin().read_line(&mut input).expect("Failed to read input.");
+        // trim the response as it includes a newline character
+        match input.trim() {
+            "increment" | "i" => {
+                // increment counter, specifying local development
+                _mock_counter.increment(Some("local".to_string()));
+            },
+            …
+    }
+    …
+```
+
+The rest of the `main()` function sets up a simple loop asking for user input. This allows us to call the functions that will be used in the blockchain without having to compile, deploy, and then invoke them.
+
+The comments explain most of the functionality, but one particular line stands out:
+
+```rust
+_mock_counter.increment(Some("local".to_string()));
+```
+
+At first glance, `Some` can be confusing. Let's look at the implementation of `increment`, the function it's calling:
+
+```rust
+pub fn increment(&mut self, debug_print: Option<String>) {
+  …
+}
+```
+
+The function `increment` takes a parameter of type `Option`. One of the explicit uses of `Option` in their docs is for ["Optional function arguments"](https://doc.rust-lang.org/std/option/) which we're doing here. 
+
+From the linked documentation, *"…every Option is either Some and contains a value, or None, and does not."* So we're calling a function that has an argument of type `Option` and giving it an explicit value of the string "local".
+
+When we eventually compile and deploy this smart contract to NEAR, the `increment()` function will get called with no arguments. While we're developing locally, we specify this argument so that it prints debugging information to our command line.
+
+<blockquote class="lesson">
+<p>Notice the naming convention of variables here. There's a lot of <code>snake_case</code> and variables prefixed with an underscore. Upon building a project, Rust informs you if naming conventions are off. It's actually quite easy to write proper Rust code using the compiler's suggestions.</p>
+</blockquote>
+
+### Our core logic: the `struct`
+
+We declare our `Counter` and `impl`, defining the functions we'll be invoking on the blockchain.
+
+Above the definitions we see attributes specific to NEAR:
+
+```rust
+#[near_bindgen]
+#[derive(Default, BorshDeserialize, BorshSerialize)]
+#[derive(Serialize, Deserialize)]
+```
+
+These essentially allow the compilation into WebAssembly to be compatible and optimized for the NEAR blockchain.
+
```

Comment:
i think the code would be cleaner and easier to reason about if you made a local `_log_helper` function that encapsulates this conditional debug output

Reply:
derp, how did I not see this. I appreciate your code reviews a lot. working on this now.